Change default epsilons and add warning in the doc. Epsilon as large …

cottrell opened the pull request:
…as 1e-3 leads to silent failures.

Redo of #33432

I think I was branched off rc2 before.

cottrell committed:
Change default epsilons and add warning in the doc. Epsilon as large as 1e-3 leads to silent failures.

robieta commented:
I don't think that we can change the default, since it would be a breaking change and also result in different behavior compared to keras-team/keras, tf.layers, and TF 1.x. However I am fine with updating the docstring to point out the effect of an epsilon which is too high.

cottrell commented:
Updating the docs to warn all future humans of the dangers sounds reasonable.

Commit:
Set default epsilon back to 1e-3

Reviewer edit: Update PR to be a pure docs chance based on review discussion. (In the interest of getting it merged.)

robieta commented:
We have decided to decline this PR on the basis that it is not relevant to enough people to justify adding it to the docstring.
